### PR TITLE
ci: Depend on init-enterprise for gen-cue verify

### PR DESCRIPTION
**Pull request by sdboyer**
**What this PR does / why we need it**:

This makes the `verify_gen_cue_step` depend on the cloning enterprise steps in the appropriate pipelines, following the same pattern as `wire_install_step`. It also makes `wire_install_step` formally depend on `verify_gen_cue_step`.

i feel like there was another thing we talked about doing, @dsotirakis? can't recall now :/

**Which issue(s) this PR fixes**:

No issue posted for it, but [build failures like this](https://drone.grafana.net/grafana/grafana-private-mirror/1352/5/14) are what we're aiming to fix. Those will start mattering when we cut the next release after 9.0.0.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by dsotirakis — CHANGES_REQUESTED** [review]
You have to:
1. Run `make drone`
2. `make drone` will fail with an `invalid dependencies` or something, because `wire-install` depends on `verify-gen-cue`, but this step runs after `wire-install`, so it fails. What you have to do I think is to include `verify-gen-cue` in the `init_steps` inside the `*.star` files **before** `wire-install`.
Again, not sure if we have to add this dependency now.

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -69,6 +69,9 @@ def wire_install_step():
         'commands': [
             'make gen-go',
         ],
+        'depends_on': [
+            'verify-gen-cue',
+        ],
```
[the remark this answers is not in the document]
Comment (sdboyer) [reply scripts/drone/steps/lib.star:74]: Yeah, i'm not worried about stability of this particular makefile directive. I've run it hundreds of times, and others have as well, and the behavior is pretty straightforward.

**Comment by sdboyer** [comment]
Derp, i always forget `make drone` 😠

**Review by dsotirakis — APPROVED** [review]
LGTM! Let's see that in action!

**Pull request merged**